### PR TITLE
Stock Price Checker: Change "form data" to "query parameter"

### PR DESCRIPTION
**Pull request by SethFalco**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

This also impacts the Spanish and Chinese file as well.
Do those need to be updated as well, or will it get caught my CrowdIn automatically if merged?

---

The project sends the NASDAQ stock ticker as a _query parameter_, not as form data. This updates the user stories to reflect this.

### CODE REVIEW

File: curriculum/challenges/english/09-information-security/information-security-projects/stock-price-checker.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ async (getUserInput) => {
 };
 ```
 
-You can send a `GET` request to `/api/stock-prices` with form data containing a NASDAQ stock ticker. Returned will be an object with the property `stockData`.
+You can send a `GET` request to `/api/stock-prices` with the query parameter `stock` representing a NASDAQ stock ticker. Returned will be an object with the property `stockData`.
```
Comment (RandellDawson) [inline curriculum/challenges/english/09-information-security/information-security-projects/stock-price-checker.md:65]:
```suggestion
You can send a `GET` request to `/api/stock-prices`, passing a NASDAQ stock symbol to a `stock` query parameter. Returned will be an object with the property `stockData`.
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@SethFalco What do you think about my suggestion?

File: curriculum/challenges/english/09-information-security/information-security-projects/stock-price-checker.md (hunk 1 above)
Comment (SethFalco) [reply curriculum/challenges/english/09-information-security/information-security-projects/stock-price-checker.md:65]: Regarding the first sentence, I feel indifferent, so I think it's fine.
I think the second sentence is much better now.

While we're suggestion other changes, may I suggest linking `NASDAQ stock symbol` to the Wikipedia page for [Ticker symbol](https://en.wikipedia.org/wiki/Ticker_symbol)?

English: https://en.wikipedia.org/wiki/Ticker_symbol
Spanish: https://es.wikipedia.org/wiki/S%C3%ADmbolo_burs%C3%A1til
Chinese: https://zh.wikipedia.org/wiki/%E8%82%A1%E7%A5%A8%E4%BB%A3%E8%99%9F

I say this as someone that didn't know what a NASDAQ ticker/stock symbol was until looking.

Comment (SethFalco) [reply curriculum/challenges/english/09-information-security/information-security-projects/stock-price-checker.md]: Actually sorry, I do have a question/comment.

With "NASDAQ stock ticker" changing to "NASDAQ stock symbol", I support the change as it's the term used on Wikipedia, but I think the language of this file should be standardized.

There is one other place that uses the term "stock ticker", shall I change this to "stock symbol" as well?

> The `stockData` property includes the `stock` ticker as a string, the price as a number, and `likes` as a number.

would become 

> The `stockData` property includes the `stock` symbol as a string, the price as a number, and `likes` as a number.

**Comment by SethFalco** [comment]
Excuse that hiccup, I accidentally pulled commits into my PR instead of rebasing. ^-^' 

I opted to apply both suggestions I had above to close the initial feedback that you had as I committed your suggestion.
Please do let me know if there're any problems with them.

**Comment by RandellDawson** [comment]
After thinking a bit more about this, I don't think it is necessary to provide a link to Wikipedia for "stock symbol".  In general, we prefer to limit external links on the challenges where possible.  I believe a user can perform the necessary research to determine what a stock symbol is.  @nhcarrigan @moT01 @scissorsneedfoodtoo ?

**Comment by nhcarrigan** [comment]
I agree. I'm all for avoiding external links whenever possible, and an important skill in development is being able to research concepts you are unfamiliar with.

**Comment by moT01** [comment]
I don't think the link should be included. See [this thread](https://github.com/freeCodeCamp/freeCodeCamp/issues/40749) for more. I'm not sure any additional information is needed here either. The demo project shows some example requests and it's a pretty easy lookup. I wouldn't be opposed to a quick note in the instructions.

File: curriculum/challenges/english/09-information-security/information-security-projects/stock-price-checker.md [hunk 2]
```diff
@@ -62,7 +62,7 @@ async (getUserInput) => {
 };
 ```
 
-You can send a `GET` request to `/api/stock-prices` with form data containing a NASDAQ stock ticker. Returned will be an object with the property `stockData`.
+You can send a `GET` request to `/api/stock-prices`, passing a NASDAQ [stock symbol](https://en.wikipedia.org/wiki/Ticker_symbol) to a `stock` query parameter. The returned object will contain a property named `stockData`.
```
Comment (RandellDawson) [inline curriculum/challenges/english/09-information-security/information-security-projects/stock-price-checker.md]: Just to be clear, the following is my suggestion for this sentence.
```suggestion
You can send a `GET` request to `/api/stock-prices`, passing a NASDAQ stock symbol to a `stock` query parameter. The returned object will contain a property named `stockData`.
```

**Comment by SethFalco** [comment]
Fair enough, I've removed the link to the external resource.
For now, I've opted not to add a note, since as @niccololampa and @moT01 suggest, develops could just look it up themselves, it's not particular difficult.

---

And sorry @RandellDawson, I already did the change! ^-^'

**Review by naomi-lgbt — APPROVED** [review]
I think I'm happy with this. Looks good to me.

**Review by RandellDawson — APPROVED** [review]

**Comment by RandellDawson** [comment]
@SethFalco Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 📝

**Pull request merged**